### ISSUE TITLE
'T' doesn't conform to the bound 'Object' of the type parameter 'T'. Try using a type that is or is a subclass of 'Object'.

### ISSUE THREAD

**Issue by divan** [oi-narendra/multiselect-dropdown#153]
Finally decided to upgrade to 3.x.x version. I'm using multidropdown in my own wrappers (for styling and convenience) and they all started throwing error:

**'T' doesn't conform to the bound 'Object' of the type parameter 'T'. Try using a type that is or is a subclass of 'Object'.**

The simplest possible to reproduce it:
```dart
class MyMultiSelect<T> extends StatelessWidget {
  final List<T> items;

  MyMultiSelect(this.items, {super.key});

  final controller = MultiSelectController<T>();

  @override
  Widget build(BuildContext context) {
    return MultiDropdown<T>(
      items: toDropdownItems(items),
      controller: controller,
    );
  }

  List<DropdownItem<T>> toDropdownItems(List<T> items) {
    return items
        .map((e) => DropdownItem(
              label: e.toString(),
              value: e,
            ))
        .toList();
  }
}
```

It turns out, MultiDropdown widget specifies T as `T extends Object`. That why `T` in my class becomes not compatible. It doesn't make sense to me, given that every class supposedly extends Object class anyway, but this is breaking widgets like mine. The solution is to add `extends Object` to the wrapper class:

```dart
class MyMultiSelect<T extends Object> extends StatelessWidget {
```
(and in the State class as well for stateful widgets).

Not sure why MultDropdown needs this 'extends Object', but this solution worked for me.

PS. Upgrading library and renaming everything and hoping that it will just work for library users is simply nuts. 3.x.x should've been a new package as it has little to do with 2.x.x.

**Comment by singgihmardianto** [comment]
I thinks it's because they also use a generic type that extends `Object` so basically you need to pass any type that extends `Object`.

**Comment by divan** [comment]
@singgihmardianto yes, but why is it needed? Every type in dart already extending Object by definition.

The only reason I can think of is for not allowing null to be passed, but I'm struggling to undertand the real world scenario where this is something that real humans would write.

I have many custom wrappers around different widgets for my form fields, and this one is the only one that requires `extends Object` to be specified in order to compile (and it's not documented anywhere).

**Comment by oi-narendra** [comment]
Closing as stale/question. If this is still an issue, please reopen with details.
